File: Week_01/G20200343040215/LeetCode_283_0215.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    if(!nums || nums.length<=1){
        return nums;
    }
    let indexNoneZero = 0;
    for(let i = 0; i< nums.length; i++){
        if(0!=nums[i]){
            let tep = nums[i];
            nums[i]=0;
            nums[indexNoneZero++] = tep;
        }            
    }
    return nums;

};

File: Week_01/G20200343040215/LeetCode_283_0215.java
package Week_01;

public class LeetCode_283_0215 {
    public void moveZeroes(int[] nums) {
        int j=0;
        for (int i=0;i<nums.length;i++){
            if (nums[i]!=0) {
                if (j!=i) {//判断
                    nums[j] = nums[i];
                    nums[i] = 0;
                }
                j++;
            }
        }
    }
}
